tiledb/cloud/taskgraphs/_codec.py

```diff
@@ -71,3 +130,102 @@ def _replace_sentinel(
     "native": cloudpickle.loads,
     "python_pickle": cloudpickle.loads,
 }
+
+
+@attrs.define(frozen=True, slots=False)
+class BinaryResult(TDBJSONEncodable):
+    """Container for a binary-encoded value, decoded on-demand.
+
+    This is used to store results obtained from the server, such that it is not
+    necessary to decode them between stages, and they only are decoded upon
+    request.
+    """
+
+    format: str
+    """The format of the data."""
```

Comment:
How about:

The TileDB format of the data (see `_MIME_TO_FORMAT`)

Reply:
Added detail.